doc/src/extras.rst

```diff
@@ -141,6 +141,262 @@ Logging cursor
+
+       Drop streaming replication slot.
+
+       :param slot_name: name of the replication slot to drop
+
+       Example::
+
+           cur.drop_replication_slot("testslot")
+
+    .. method:: start_replication(slot_type, slot_name=None, writer=None, start_lsn=None, timeline=0, keepalive_interval=10, options=None)
+
+       Start a replication stream.  On non-asynchronous connection, also
+       consume the stream messages.
+
+       :param slot_type: type of replication: either `REPLICATION_PHYSICAL` or
+                         `REPLICATION_LOGICAL`
+       :param slot_name: name of the replication slot to use (required for
+                         logical replication)
+       :param writer: a file-like object to write replication messages to
+       :param start_lsn: the LSN position to start from, in the form
+                         ``XXX/XXX`` (forward-slash separated pair of
+                         hexadecimals)
+       :param timeline: WAL history timeline to start streaming from (optional,
+                        can only be used with physical replication)
+       :param keepalive_interval: interval (in seconds) to send keepalive
+                                  messages to the server
+       :param options: a dictionary of options to pass to logical replication
+                       slot
+
+       When used on non-asynchronous connection this method enters an endless
+       loop, reading messages from the server and passing them to ``write()``
+       method of the *writer* object.  This is similar to operation of the
+       `~cursor.copy_to()` method.  It also sends keepalive messages to the
+       server, in case there were no new data from it for the duration of
+       *keepalive_interval* seconds (this parameter's value must be equal to
+       at least than 1 second, but it can have a fractional part).
+
+       With asynchronous connection, this method returns immediately and the
+       calling code can start reading the replication messages in a loop.
+
+       A sketch implementation of the *writer* object for logical replication
+       might look similar to the following::
+
+           from io import TextIOBase
+
+           class LogicalStreamWriter(TextIOBase):
+
+               def write(self, msg):
+                   self.store_message_data(msg.payload)
+
+                   if self.should_report_to_the_server_now(msg):
+                       msg.cursor.send_replication_feedback(flush_lsn=msg.wal_end)
+
+       First, like with the `~cursor.copy_to()` method, the code that calls
+       the provided ``write()`` method checks if the *writer* object is
+       inherited from `~io.TextIOBase`.  If that is the case, the message
+       payload to be passed is converted to unicode using the connection's
+       `~connection.encoding` information.  Otherwise, the message is passed
+       as is.
+
+       The *msg* object being passed is an instance of `~ReplicationMessage`
+       class.
```

Comment:
> The *msg* object being passed is an instance of `~ReplicationMessage` class.

This should be higher up, since it's confusing to read and understand the code above w/o knowing that.